src/wasm-lib/kcl/src/lsp/kcl/mod.rs

```diff
@@ -285,6 +288,79 @@ impl Backend {
 
         completions
     }
+
+    pub async fn send_telemetry(&self) -> Result<()> {
+        // Get information about the user.
+        let user = self
+            .zoo_client
+            .users()
+            .get_self()
+            .await
+            .map_err(|e| anyhow::anyhow!(e.to_string()))?;
+
+        // Hash the user's id.
+        // Create a SHA-256 object
+        let mut hasher = sha2::Sha256::new();
+        // Write input message
+        hasher.update(user.id);
+        // Read hash digest and consume hasher
+        let result = hasher.finalize();
+        // Get the hash as a string.
+        let user_id_hash = format!("{:x}", result);
+
+        // Get the workspace folders.
+        // The key of the workspace folder is the project name.
+        let workspace_folders = self.workspace_folders();
+        let project_names: Vec<String> = workspace_folders.iter().map(|v| v.name.clone()).collect::<Vec<_>>();
+        // Get the first name.
+        let project_name = project_names
+            .first()
+            .ok_or_else(|| anyhow::anyhow!("no project names"))?
+            .to_string();
+
+        // Collect all the file data we know.
+        let mut attachments = vec![];
+        for entry in self.current_code_map.iter() {
+            let file_name = entry.key().replace("file://", "").to_string();
+
+            let content_type = Some(mime_guess::from_path(&file_name).first_or_octet_stream().to_string());
+
+            // Try to get the content type from the file name.
+            attachments.push(kittycad::types::multipart::Attachment {
+                // Clean the URI part.
+                name: file_name.clone(),
```

Comment:
The server currently expects a single attachment with the name "attachment". The plan was to zip client-side. But, it would also be easy to just send each file like you are now then zip server-side.

Reply:
oh ohhhh, let me hook up a zip lib

Reply:
okay wait sorry this gets weird
because basically I have this interface where I call read/readir/exists on files which tehn translates to js, so id essentially have to write tthe files back out, but i cant write to disk, id need a whole interface for that, actaully thats stupid i guess in javascript i could zip them up, anyways sorry my point is i cant do this on the rust side since i dont actually have a real file interface just a fake js one, and so it gets nasty